### ISSUE TITLE
oc cluster up => down => up failed if service catalog flag is used.

### ISSUE THREAD

**Issue by praveenkumar** [openshift/origin#18119]
If a user try to use `use-existing-config` for service-catalog and do `cluster up` => `cluster down` then next `cluster up` failed.

##### Version
```
$ oc version
oc v3.7.0+7ed6862
kubernetes v1.7.6+a08f5eeb62
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://127.0.0.1:8443
openshift v3.7.0+7ed6862
kubernetes v1.7.6+a08f5eeb62
```

##### Steps To Reproduce
```
$ oc cluster up --use-existing-config --version v3.7.0 --service-catalog --host-config-dir /tmp/minishift/openshift.local.config --host-data-dir /tmp/minishift/hostdata --host-pv-dir /tmp/minishift/openshift.local.pv --host-volumes-dir /tmp/minishift/openshift.local.volumes
Starting OpenShift using openshift/origin:v3.7.0 ...
OpenShift server started.

The server is accessible via web console at:
    https://127.0.0.1:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

$ oc cluster down

$ oc cluster up --use-existing-config --version v3.7.0 --service-catalog --host-config-dir /tmp/minishift/openshift.local.config --host-data-dir /tmp/minishift/hostdata --host-pv-dir /tmp/minishift/openshift.local.pv --host-volumes-dir /tmp/minishift/openshift.local.volumes
Starting OpenShift using openshift/origin:v3.7.0 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v3.7.0 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... 
   WARNING: Binding DNS on port 8053 instead of 53, which may not be resolvable from all clients.
-- Checking type of volume mount ... 
   Using nsenter mounter for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ... 
   Using 127.0.0.1 as the server IP
-- Checking service catalog version requirements ... OK
-- Starting OpenShift container ... 
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Registering template service broker with service catalog ... FAIL
   Error: cannot register the template service broker
   Caused By:
     Error: cannot create objects from template openshift-infra/template-service-broker-registration
     Caused By:
       Error: an error on the server ("Error: 'dial tcp 172.30.1.2:443: getsockopt: connection refused'\nTrying to reach: 'https://172.30.1.2:443/apis/servicecatalog.k8s.io/v1beta1/clusterservicebrokers'") has prevented the request from succeeding (post clusterservicebrokers.servicecatalog.k8s.io)
```

##### Current Result

```
Caused By:
       Error: an error on the server ("Error: 'dial tcp 172.30.1.2:443: getsockopt: connection refused'\nTrying to reach: 'https://172.30.1.2:443/apis/servicecatalog.k8s.io/v1beta1/clusterservicebrokers'") has prevented the request from succeeding (post clusterservicebrokers.servicecatalog.k8s.io)
```

##### Expected Result

It should be succeed without any error.

**Comment by LorbusChris** [comment]
possibly related: https://github.com/minishift/minishift/issues/1910#issuecomment-374949404

**Comment by BenHall** [comment]
Seeing this with Openshift 3.9 too

**Comment by LalatenduMohanty** [comment]
@deads2k Is this got fixed when you guys did the refactoring for 3.10?

**Comment by deads2k** [comment]
> @deads2k Is this got fixed when you guys did the refactoring for 3.10?

I think so.  The failure mode would be different now at the very least.
